cmake/Modules/Findaom.cmake

```diff
@@ -42,9 +42,21 @@ mark_as_advanced(AOM_INCLUDE_DIR AOM_LIBRARY AOM_LIBRARIES)
 if(AOM_LIBRARY)
     if("${AOM_LIBRARY}" MATCHES "\\${CMAKE_STATIC_LIBRARY_SUFFIX}$")
         add_library(aom STATIC IMPORTED GLOBAL)
+        set(_AOM_PC_PREFIX "_AOM_STATIC")
     else()
         add_library(aom SHARED IMPORTED GLOBAL)
+        set(_AOM_PC_PREFIX "_AOM")
     endif()
     set_target_properties(aom PROPERTIES IMPORTED_LOCATION "${AOM_LIBRARY}")
     target_include_directories(aom INTERFACE ${AOM_INCLUDE_DIR})
+
+    set(_AOM_PC_LIBRARIES "${${_AOM_PC_PREFIX}_LIBRARIES}")
+    # remove "aom" so we only have library dependencies
+    list(REMOVE_ITEM _AOM_PC_LIBRARIES "aom")
+
+    # Add absolute paths to libraries
+    foreach(_lib ${_AOM_PC_LIBRARIES})
+      find_library(_aom_dep_lib_${_lib} ${_lib} HINTS ${${_AOM_PC_PREFIX}_LIBRARY_DIRS})
+      target_link_libraries(aom INTERFACE ${_aom_dep_lib_${_lib}})
+    endforeach()
```

Comment:
It seems that this kind of fix is also needed if an AV1 codec other than libaom is a static library and has dependencies, right?

This pull request only modifies the libaom-related code. Are you trying to reduce risks by only fixing issues that a customer has reported? Or do other AV1 codecs not need this kind of fix?

Reply:
It's a possibility for all of them, in theory. With rav1e, a pkg-config check could replace the hard-coded list of windows system libraries we add as dependencies. But in practice, I think aom is the only library with optional dependencies that could cause linking issues.